main/player.py

```diff
@@ -22,20 +23,17 @@ def add_to_backpack(self, item):
     def remove_from_backpack(self, item):
         if item in self.backpack:
             self.backpack.remove(item)
-            print(f"{item} removed from backpack")
-        else:
-            print(f"{item} not found in backpack")
-
-    def expand_backpack(self):
-        self.backpack_size += 1
-        print(f"Backpack size has increased to {self.backpack_size}")
 
-    def gain_experience(self, gained_experience):
-        self.experience += gained_experience
-        if self.experience >= self.experience_needed:
+    def gain_experience(self, sum_of_enemys_stats):
+        self.experience += sum_of_enemys_stats
+        while self.experience >= self.experience_needed:
```

Comment:
I'm wonder why did you use `while` loop instead of `if`. I think that will work correctly but i'm just curious.

Reply:
If statement didn't work in case when i've got more exp than for 1 lvl ( it only gave 1 lvl and seemed kinda buggy).
While loop seems to resolve this problem.